LDAR_Sim/src/ldar_sim_run.py

```diff
@@ -101,26 +100,43 @@ def ldar_sim_run(simulation, weather, daylight):
 
 
 def simulate(
-    sim_num,
-    prog,
+    daylight,
+    weather,
+    sim_num: int,
+    prog_name: str,
+    meth_params,
     sim_settings,
     virtual_world,
     infrastructure,
     input_dir,
     output_dir,
     preseed_timeseries,
 ):
+    infra = copy.deepcopy(infrastructure)
```

Comment:
Discussion: If you're going to copy the infrastructure here, (which is theoretically better than copying it over in the main scope since the memory should get free once the simulation ends), just make sure that copy.deepcopy is process/thread safe or we could see some really strange bugs. If it isn't you can still do this, but it'll need to be inside whatever the process version of a mutex is. Also maybe call the garbage collector after this copy to force an immediate cleanup of the original (might need to save it inside a data passing class so the input parameters don't keep it in scope)?

Reply:
Great catch. Deepcopy is thread safe, but it's not process safe.